scheduler/scheduler.go

```diff
@@ -48,9 +48,12 @@ type Plugin interface {
 	// NewPlugin creates a new instance of plugin
 	NewPlugin(ca security.CAAPI) Plugin
 
-	// Connect initializes the connection with the execution command
-	// and the log path wbere the logs should be stored.
-	Connect(command *exec.Cmd, logPath *string) error
+	// Init initializes the go-plugin client and generates a
+	// new certificate pair for gaia and the plugin/pipeline.
+	Init(command *exec.Cmd, logPath *string) error
```

Comment:
Hmm, do you have an example of how Init and Validate would look like? Perhaps in another branch somewhere?

If I understand this correctly, you basically replaced connect with two steps. 1. Init 2. Validate?

Reply:
Yes. The problem is `connect` did several things: 
1. Create new certificates
2. Prepares the log path/writer
3. Creates and starts the plugin
4. Validates if the plugin is compatible

It was really hard to test this one function. To somehow write valid tests for it I had to mock some things and therefore I had to split this function up into two steps. Nothing else has been changed actually.